test/functional/forall/reduce-sanity/CMakeLists.txt

```diff
@@ -5,36 +5,63 @@
 # SPDX-License-Identifier: (BSD-3-Clause)
 ###############################################################################
 
-raja_add_test(
-  NAME test-forall-reduce-sanity-seq
-  SOURCES test-forall-reduce-sanity-seq.cpp)
+#
+# Function that generates test file and build target for each
+# reduction type and each backend
+#
+function( buildforallreducesanitytest REDUCETYPES )
+  foreach( REDUCETYPE ${REDUCETYPES} )
+    configure_file( test-forall-reduce-sanity.cpp.in 
+                    test-forall-${REDUCETYPE}-sanity-${BACKEND}.cpp )
+    raja_add_test( NAME test-forall-${REDUCETYPE}-sanity-${BACKEND}
+                   SOURCES test-forall-${REDUCETYPE}-sanity-${BACKEND}.cpp )
+
+    target_include_directories(test-forall-${REDUCETYPE}-sanity-${BACKEND}.exe 
+                               PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/tests)
+  endforeach()
+endfunction()
+
+
+#
+# Variable holding reduction types
+#
+set(REDUCETYPES ReduceSum ReduceMin ReduceMax ReduceMinLoc ReduceMaxLoc)
+
+
+#
+# Generate tests for each enabled RAJA back-end
+#
+
+set( BACKEND Sequential )
+  buildforallreducesanitytest( "${REDUCETYPES}" )
+unset( BACKEND )
 
 if(RAJA_ENABLE_OPENMP)
-  raja_add_test(
-    NAME test-forall-reduce-sanity-openmp
-    SOURCES test-forall-reduce-sanity-openmp.cpp)
+  set( BACKEND OpenMP )
+    buildforallreducesanitytest( "${REDUCETYPES}" )
+  unset( BACKEND )
 endif()
```

Comment:
Can we do something like this in cmake?
```
set( BACKENDS Sequential )
if(RAJA_ENABLE_OPENMP)
  set( BACKENDS ${BACKENDS} OpenMP )
endif()
```

Reply:
You can certainly append into a list, how that gets translated into something for C++ is an extra step to think about though.

Reply:
Is the idea that we would collect all the enabled back-ends in a list and then call the test generating function in a loop over backends?

Reply:
list(APPEND BACKENDS OpenMP)

Reply:
That or have it take two lists and loop over both.

Reply:
I don't know if that would really save us much though.

Reply:
Only if we have a "backends" variable somewhere that we make once and re-use, rather than doing this for every test set, do I think it would make much difference.

Reply:
Then, we lose the ability to be selective about which tests we launch for backends. For example, we have no scan capability for OpenMP target back-end.

Reply:
Fair point.

Reply:
As long as the BACKENDS is an argument to the test generator we can have a separate SCAN_BACKENDS list.

Reply:
True. I'll think about this a bit more and see what I come up with. Thanks for the feedback.

Reply:
I've changed the workgroup tests to use this as another use case.

Reply:
Cool. I'll take a look.